### PR TITLE
Added Set-Cookie headers processing

### PR DESCRIPTION
**Pull request by dnekhayev**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	<modelVersion>4.0.0</modelVersion>
 	<groupId>com.github.hi-fi</groupId>
 	<artifactId>httpclient</artifactId>
-	<version>0.0.6-SNAPSHOT</version>
+	<version>0.0.7-SNAPSHOT</version>
```
Comment (Hi-Fi) [inline pom.xml]: No need to increase the version. Last release was 0.0.5, so next is going to be 0.0.6, not 0.0.7.

**Review by Hi-Fi — CHANGES_REQUESTED** [review]
Have you tested this locally with https://github.com/Hi-Fi/robotframework-httprequestlibrary? Is this still working as should with it, and provide the information you were needing?

File: pom.xml (hunk 1 above)
Comment (dnekhayev) [reply pom.xml]: > Have you tested this locally with https://github.com/Hi-Fi/robotframework-httprequestlibrary? Is this still working as should with it, and provide the information you were needing?

I've just tested locally. It works good for me.

**Comment by dnekhayev** [comment]
Added one more commit for backward compatibility. 
I've tested it locally. It works good for me.

**Comment by Hi-Fi** [comment]
I think this looks good, but I have to fix the CI of the repo before merging. I'll make the fix hopefully today, so you can then rebase from master with working CI.

**Review by Hi-Fi — CHANGES_REQUESTED** [review]
Can you please rebase with master, and squash commits into one.

I prepared also the plugin-repo in a way that it would be quite quick to make release of it after this one is available at Maven Central.

**Comment by dnekhayev** [comment]
> Can you please rebase with master, and squash commits into one.
> 
> I prepared also the plugin-repo in a way that it would be quite quick to make release of it after this one is available at Maven Central.

Done. Please review the changes.

File: src/main/java/com/github/hi_fi/httpclient/cookie/Cookie.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    public Cookie(String setCookieValue) {
+        parse(setCookieValue);
+    }
+
+    public String name() {
+        return size() == 0 ? "" : entrySet().iterator().next().getKey();
+    }
+
+    public String value() {
+        return size() == 0 ? "" : entrySet().iterator().next().getValue();
```
Comment (php-coder) [inline src/main/java/com/github/hi_fi/httpclient/cookie/Cookie.java:18]: By the way, if `value()` is identical to `name()` why not just invoke one from another?

Comment (dnekhayev) [reply src/main/java/com/github/hi_fi/httpclient/cookie/Cookie.java:18]: > By the way, if `value()` is identical to `name()` why not just invoke one from another?

Hi,
I am not quite sure that I understand you. 
Actually, name() and value() are different: getKey() vs getValue() respectively.
These values are derived from Set-Cookie header's value by splitting it into 2 parts by first '=' symbol.

Comment (php-coder) [reply src/main/java/com/github/hi_fi/httpclient/cookie/Cookie.java:18]: > Actually, name() and value() are different: getKey() vs getValue() respectively.

I see it now, thanks! Then disregard my comment.

**Pull request merged**